libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Consensus.hs

```diff
@@ -0,0 +1,1166 @@
+          (Coin 100)
+    Conway _ ->
+      Wdrl $
+        Map.singleton
+          (RewardAcnt Testnet (keyToCredential exampleStakeKey))
+          (Coin 100)
+
+-- | These are dummy values.
+testShelleyGenesis :: ShelleyGenesis era
+testShelleyGenesis =
+  ShelleyGenesis
+    { sgSystemStart = UTCTime (fromGregorian 2020 5 14) 0,
+      sgNetworkMagic = 0,
+      sgNetworkId = Testnet,
+      -- Chosen to match activeSlotCoeff
+      sgActiveSlotsCoeff = unsafeBoundRational 0.9,
+      sgSecurityParam = securityParameter testGlobals,
+      sgEpochLength = runIdentity $ epochInfoSize testEpochInfo 0,
+      sgSlotsPerKESPeriod = slotsPerKESPeriod testGlobals,
+      sgMaxKESEvolutions = maxKESEvo testGlobals,
+      -- Not important
+      sgSlotLength = secondsToNominalDiffTime 2,
+      sgUpdateQuorum = quorum testGlobals,
+      sgMaxLovelaceSupply = maxLovelaceSupply testGlobals,
+      sgProtocolParams = emptyPParams,
+      sgGenDelegs = Map.empty,
+      sgInitialFunds = mempty,
+      sgStaking = emptyGenesisStaking
+    }
+
+testEpochInfo :: EpochInfo Identity
+testEpochInfo = epochInfoPure testGlobals
+
+-- ================================================================================
+
+-- | Build a Tx from a list of maybe fields. The maybe is usfull because sometimes we
+--   only want to add a field in a particular era, so we might say something like this
+-- genericTx proof
+--  [Just field1
+--  , if (Some proof) >= Mary Mock then Just field2 else Nothing
+--  , Just field3
+--  ]
+genericTx :: Proof era -> [Maybe (TxField era)] -> Core.Tx era
+genericTx proof xs = newTx proof (foldl' accum [] xs)
+  where
+    accum ans Nothing = ans
+    accum ans (Just field) = field : ans
+
+-- | Build a Tx from a list of maybe fields. The maube is usfull because sometimes we
+--   only want to add a field in a particular era, so we might say something like this
+-- genericTxBody proof
+--  [Just field1
+--  , if (Some proof) >= Mary Mock then Just field2 else Nothing
+--  , Just field3
+--  ]
+genericTxBody :: Era era => Proof era -> [Maybe (TxBodyField era)] -> Core.TxBody era
+genericTxBody proof xs = specialize @Core.EraTxBody proof (newTxBody proof (foldl' accum [] xs))
+  where
+    accum ans Nothing = ans
+    accum ans (Just field) = field : ans
+
+genericTxOut :: Era era => Proof era -> [Maybe (TxOutField era)] -> Core.TxOut era
+genericTxOut proof xs = newTxOut proof (foldl' accum [] xs)
+  where
+    accum ans Nothing = ans
+    accum ans (Just field) = field : ans
+
+genericWits :: Era era => Proof era -> [Maybe (WitnessesField era)] -> Core.TxWits era
+genericWits proof xs = newWitnesses merge proof (foldl' accum [] xs)
+  where
+    accum ans Nothing = ans
+    accum ans (Just field) = field : ans
+
+-- =========================================================
+-- Individual examples for each Era
+-- =========================================================
+
+type StandardShelley = ShelleyEra StandardCrypto
```

Comment:
They don't clash with Proofs, Proofs are at the value level, while type synonyms are at the type level.

We don't want to have every single package redefining theses type synonyms. I removed 5 different copies already throughout tests suites, all with different names.

I created those on purpose, because we need to be consistent. They are also exported in the API.

If you not gonna delete them, I'll have to do that later. Please Tim, delete these duplicate type synonyms

Reply:
I inlined the type synonyms. They each had just one use.